### PR TITLE
[Internal] CosmosClient: Adds new internal constructor

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

1. Adds a new internal CosmosClient constructor needed for compute to handle resource token scenarios where they need to pass in a custom AuthorizationTokenProvider.
2. This simplifies the Constructors to call the same base implementation to remove duplicate code.

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs [hunk 1]
```diff
@@ -239,25 +222,34 @@ protected CosmosClient()
...
             {
                 throw new ArgumentNullException(nameof(accountEndpoint));
             }
 
-            if (tokenCredential == null)
-            {
-                throw new ArgumentNullException(nameof(tokenCredential));
-            }
+            this.Endpoint = new Uri(accountEndpoint);
+            this.AuthorizationTokenProvider = authorizationTokenProvider ?? throw new ArgumentNullException(nameof(authorizationTokenProvider));
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/CosmosClient.cs:249]: Might be hard but anyway to validate this gets called? Maybe Mock or create a simple implementation to verify it gets used when passed?

**Review by ealsur — APPROVED** [review]
LGTM, please see if we can add tests to confirm it gets used.

**Review by FabianMeiswinkel — DISMISSED** [review]
LGTM

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs (hunk 1 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/CosmosClient.cs:249]: I added a new UT to validate it.

**Review by ealsur — APPROVED** [review]

**Pull request merged**